rd-net/RdFramework/Impl/SocketWire.cs

```diff
@@ -639,25 +643,32 @@ public Server(Lifetime lifetime, IScheduler scheduler, Socket serverSocket, stri
       private Server(Lifetime lifetime, IScheduler scheduler, string? optId = null) : base("ServerSocket-"+(optId ?? "<noname>"), lifetime, scheduler)
       {}
 
-      public static Socket CreateServerSocket(IPEndPoint? endPoint)
+      public static Socket CreateServerSocket(EndPointWrapper? endPointWrapper)
       {
-        Protocol.InitLogger.Verbose("Creating server socket on endpoint: {0}", endPoint);
+        Protocol.InitLogger.Verbose("Creating server socket on endpoint: {0}", endPointWrapper?.EndPointImpl);
+        // by default we will use IPEndpoint?
+        endPointWrapper ??= EndPointWrapper.CreateIpEndPoint(IPAddress.Loopback, 0);
 
-        var serverSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+        var serverSocket = new Socket(endPointWrapper.AddressFamily, endPointWrapper.SocketType, endPointWrapper.ProtocolType);
         SetSocketOptions(serverSocket);
 
-        endPoint = endPoint ?? new IPEndPoint(IPAddress.Loopback, 0);
-        serverSocket.Bind(endPoint);
+        if (endPointWrapper.LocalPath != null)
+        {
+          if (File.Exists(endPointWrapper.LocalPath)) File.Delete(endPointWrapper.LocalPath);
+        }
```

Comment:
A question: shouldn't we expect that the file is delected by the caller? (Not really sure what is the practical answer; I'd just prefer to not delete anything unless really required.)